src/client.c

```diff
@@ -221,9 +221,22 @@ static bool login_via_cert(PgSocket *client)
 		goto fail;
 
 	log_debug("TLS cert login: %s", tls_peer_cert_subject(client->sbuf.tls));
+
 	if (!tls_peer_cert_contains_name(client->sbuf.tls, client->login_user->name)) {
-		slog_error(client, "TLS certificate name mismatch");
-		goto fail;
+		log_debug("TLS certificate name mismatch. Looking for a map.");
```

Comment:
This logic is incorrect in a small but very important way. W should only try a direct match if no map is defined. If a map is defined, it's very much possible that a direct match is actually not allowed to give access.